### PR TITLE
Reload Secure Settings REST specs & docs

### PR DESCRIPTION
**Pull request by albertzaharovits**
This is a minimal REST API spec and docs for the REST handler for the `_nodes/reload_secure_settings` endpoint.

Relates https://github.com/elastic/elasticsearch/issues/29135

### CODE REVIEW

**Review by jaymode — CHANGES_REQUESTED** [review]
The password has been removed from this API so we need to fix those docs. I also added the 6.4.1 label to this since we need this on the 6.4 branch too

File: docs/reference/cluster/nodes-reload-secure-settings.asciidoc [hunk 1]
```diff
@@ -0,0 +1,57 @@
+[[cluster-nodes-reload-secure-settings]]
+== Nodes Reload Secure Settings
+
+The cluster nodes reload secure settings API is used to re-read the
+local node's encrypted keystore. Specifically, it broadcasts a password
```
[the remark this answers is not in the document]
Comment (albertzaharovits) [reply docs/reference/cluster/nodes-reload-secure-settings.asciidoc]: Hah... I have checked the code of course, and I felt something was missing, I recalled adding the password in the request body, yet now it wasn't there, though I haven't investigated further. Thanks for the pointer Jay!

There is a note after this paragraph, disclaiming the password broadcast statement:
```
Note: At the moment, the password parameter is not supported. The empty password
is the only valid value. Consequently, the request body is empty.
```

I presume you have noticed it, but, in order to be consistent with the pointed change, you wish to banish the mention of 'password'.
I will make the changes.

**Comment by albertzaharovits** [comment]
@jaymode is this satisfactory? or more thorough testing is required? I would say it's decent as it is.

**Review by jaymode — APPROVED** [review]
LGTM

**Pull request merged**